### PR TITLE
Adding AI Exception tracking

### PR DESCRIPTION
**Pull request by ryuyu**
Adding a middleware that tracks our exception as AI exceptions.

@skofman1 @joelverhagen @shishirx34

### CODE REVIEW

File: src/NuGet.Services.BasicSearch/Startup.cs [hunk 1]
```diff
@@ -314,6 +317,7 @@ private void TrackIndexMetrics(NuGetSearcherManager searcherManager, SearchTelem
             catch (Exception e)
             {
                 await _responseWriter.WriteResponseAsync(context, e, _logger);
+                throw e;
```
Comment (shishirx34) [inline src/NuGet.Services.BasicSearch/Startup.cs]: There is ClientException catch above this block, when are these thrown? Do we want to log it?

Comment (joelverhagen) [inline src/NuGet.Services.BasicSearch/Startup.cs]: Don't `throw e;`. `throw;`

File: src/NuGet.ApplicationInsights.Owin/ExceptionTrackingMiddleware.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.ApplicationInsights;
```
Comment (joelverhagen) [inline src/NuGet.ApplicationInsights.Owin/ExceptionTrackingMiddleware.cs]: Sort usings

Comment (joelverhagen) [inline src/NuGet.ApplicationInsights.Owin/ExceptionTrackingMiddleware.cs]: System first

**Pull request merged**